src/app/app.component.ts

```diff
@@ -208,6 +209,99 @@ export class AppComponent implements OnInit {
       this.schema,
       this.dataSources.dataSources
     );
+
+    // Get concepts with no label
+    const concepts = this.traverseForUnlabeledConcepts(this.form.rootNode);
+
+    // Fetch missing labels from concept dictionary
+    this.fetchMockedConceptData(concepts).then((conceptData: any) => {
+      this.labelMap = [];
+      conceptData.forEach((concept: any) => {
+        this.labelMap[concept.reqId] = concept.display;
+      });
+    });
+  }
+
+  fetchMockedConceptData (concepts) {
+    const promise = new Promise(function(resolve, reject) {
+      // Simulate a server response with some delay
+      setTimeout(function() {
+        const conceptData = [
+          {
+            "uuid": "a89ff9a6-1350-11df-a1f1-0026b9348838",
+            "display": "Was this visit scheduled?",
+            "reqId": concepts[0]
```

Comment:
The "reqId" is used to identify the concept because the request for its data can be made using the same-as-mapping instead of the uuid and then we would have no way of matching the retrieved concept data with the request concept string